### PR TITLE
API Remove Object class

### PR DESCRIPTION
**Pull request by tractorcow**
I've also updated DataObjectSchema::manyManyComponent() return array is now associative array, because the code was gross and using `list()` assignment had lots of linting issues.

Parent issue https://github.com/silverstripe/silverstripe-framework/issues/6085

### CODE REVIEW

File: src/View/Parsers/Transliterator.php [hunk 1]
```diff
@@ -53,11 +57,11 @@ protected function useStrTr($source)
             'Õ'=>'O', 'Ö'=>'Oe', 'Ø'=>'O', 'Ù'=>'U', 'Ú'=>'U', 'Û'=>'U', 'Ü'=>'Ue', 'Ý'=>'Y', 'Þ'=>'B', 'ß'=>'ss',
             'à'=>'a', 'á'=>'a', 'â'=>'a', 'ã'=>'a', 'ä'=>'ae', 'å'=>'a', 'æ'=>'ae', 'ç'=>'c', 'è'=>'e', 'é'=>'e',
             'ê'=>'e', 'ë'=>'e', 'ì'=>'i', 'í'=>'i', 'î'=>'i', 'ï'=>'i', 'ð'=>'o', 'ñ'=>'n', 'ò'=>'o', 'ó'=>'o',
-            'ô'=>'o', 'õ'=>'o', 'ö'=>'oe', 'ø'=>'o', 'ù'=>'u', 'ú'=>'u', 'û'=>'u', 'ü'=>'ue', 'ý'=>'y', 'ý'=>'y',
+            'ô'=>'o', 'õ'=>'o', 'ö'=>'oe', 'ø'=>'o', 'ù'=>'u', 'ú'=>'u', 'û'=>'u', 'ü'=>'ue', 'ý'=>'y',
             'þ'=>'b', 'ÿ'=>'y', 'Ŕ'=>'R', 'ŕ'=>'r',
             'Ā'=>'A', 'ā'=>'a', 'Ē'=>'E', 'ē'=>'e', 'Ī'=>'I', 'ī'=>'i', 'Ō'=>'O', 'ō'=>'o', 'Ū'=>'U', 'ū'=>'u',
-            'œ'=>'oe', 'ß'=>'ss', 'ĳ'=>'ij', 'ą'=>'a','ę'=>'e', 'ė'=>'e', 'į'=>'i','ų'=>'u','ū'=>'u', 'Ą'=>'A',
-            'Ę'=>'E', 'Ė'=>'E', 'Į'=>'I','Ų'=>'U','Ū'=>'U',
+            'œ'=>'oe', 'ĳ'=>'ij', 'ą'=>'a','ę'=>'e', 'ė'=>'e', 'į'=>'i', 'ų'=>'u', 'Ą'=>'A',
```
Comment (flamerohr) [inline src/View/Parsers/Transliterator.php]: how come these values have changed in the pull request?

File: src/Forms/FormRequestHandler.php [hunk 2]
```diff
@@ -253,7 +253,7 @@ public function httpSubmission($request)
         if ($legacyActions) {
             throw new BadMethodCallException(
                 "allowed_actions are not valid on Form class " . get_class($this->form) .
-                ". Implement these in subclasses of " . get_class($this) . " instead"
+                ". Implement these in subclasses of " . static::class . " instead"
```
Comment (tractorcow) [inline src/Forms/FormRequestHandler.php]: I used auto-replacement for many of these.

File: src/ORM/DataObject.php [hunk 3]
```diff
@@ -2914,12 +2900,12 @@ public static function get_one($callerClass, $filter = "", $cache = true, $order
...
     public function flushCache($persistent = true)
     {
-        if ($this->class == self::class) {
+        if (static::class == self::class) {
             self::$_cache_get_one = array();
             return $this;
         }
 
-        $classes = ClassInfo::ancestry($this->class);
+        $classes = ClassInfo::ancestry(static::class);
```
Comment (tractorcow) [inline src/ORM/DataObject.php]: class name is ok.

File: src/ORM/DataObject.php [hunk 4]
```diff
@@ -1605,7 +1596,7 @@ public function getComponents($componentName)
         $joinField = $schema->getRemoteJoinField(static::class, $componentName, 'has_many', $polymorphic);
         /** @var HasManyList $result */
         if ($polymorphic) {
-            $result = PolymorphicHasManyList::create($componentClass, $joinField, $this->class);
+            $result = PolymorphicHasManyList::create($componentClass, $joinField, get_class($this));
```
Comment (tractorcow) [inline src/ORM/DataObject.php]: Yes I'll auto-replace these.

File: src/View/Parsers/Transliterator.php (hunk 1 above)
Comment (tractorcow) [inline src/View/Parsers/Transliterator.php]: PHP linting issues with duplicate keys.

File: .travis.yml [hunk 5]
```diff
@@ -56,6 +56,9 @@ before_script:
   - phpenv config-rm xdebug.ini
   - echo 'memory_limit = 2048M' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
 
+# Temporarily update to 1.5.x-dev of composer
+  - composer self-update --snapshot
```
Comment (flamerohr) [inline .travis.yml]: @tractorcow pinging you for future reference to remove when composer stops breaking and we don't need the nightly build anymore

**Pull request merged**